pkg/processor/trigger/kafka/trigger.go

```diff
@@ -519,3 +573,47 @@ func (k *kafka) signalWorkerTermination(workerTerminationCompleteChan chan bool)
 	// signal termination complete
 	workerTerminationCompleteChan <- true
 }
+
+// subscribeToControlMessageKind subscribes all workers to control message kind
+func (k *kafka) subscribeToControlMessageKind(kind controlcommunication.ControlMessageKind,
+	controlMessageChan chan *controlcommunication.ControlMessage) error {
+
+	k.Logger.DebugWith("Subscribing to control message kind", "kind", kind)
+
+	for _, workerInstance := range k.WorkerAllocator.GetWorkers() {
+		if err := workerInstance.Subscribe(kind, controlMessageChan); err != nil {
+			return errors.Wrapf(err, "Failed to subscribe to explicit ack control message kind in worker %d", workerInstance.GetIndex())
+		}
+	}
+
+	return nil
+}
+
+// explicitAckHandler reads offset data messages from the trigger's control channel, and marks the
+// offset accordingly
+func (k *kafka) explicitAckHandler(session sarama.ConsumerGroupSession,
+	controlMessageChan chan *controlcommunication.ControlMessage,
+	ackWindowSize int64) {
+
+	k.Logger.DebugWith("Listening to explicit ack control messages", "ackWindowSize", ackWindowSize)
```

Comment:
i wouldnt consider the ack window size when user explicitly acked.
the window size is a different feature

Reply:
Marking the offset uses the `ackWindowSize`, as seen here:
https://github.com/nuclio/nuclio/pull/2627/files/ecad0e8b7b26cb35318863d4550f75a0cbd92ec1#diff-398bd9db5a2ffa7ec3408efc766fb73ae9de25113ca300bae962874b32ed04caR615

Do you want to ignore it in the case of explicit ack?

Reply:
i think so. comment ack window usage in case of explicit ack.